src/main/kotlin/uk/co/poolefoundries/baldinggate/core/CoreComponents.kt

```diff
@@ -72,6 +73,21 @@ data class PositionComponent(val x: Int, val y: Int) : Component {
 
 }
 
+data class AnimationComponent(val entity: Entity, var positions: List<PositionComponent> = listOf<PositionComponent>(), var progress: Float = 0F, var isFinished: Boolean = false) : Component {
+    val animationDuration = 0.1F
+
+    fun animationStep(delta: Float) {
+        this.progress += delta / animationDuration
+        val positionIndex = round(this.progress).toInt()
+        if (positionIndex >= this.positions.size-1){
```

Comment:
Could do: `val positionIndex = min(round(this.progress).toInt(), this.positions.size-1)`

And have a 
```kotlin 
fun isFinsihed() : Bool {
   return progress > this.positions.size * animationDuration
} 
```